### PR TITLE
Use a fixed position context view as a child of the anchor element

### LINKED ISSUE #90729: Escaping from a custom menu loses focus
1 Tab to this
image
2 Press Enter
image
3 Press Esc

Expected: focus returns to '...'
Observed: '...' button disappears and the focus moves to somewhere invisible.
image

V 1.42.1 on Windows 7.

### PR DESCRIPTION
**Pull request by sbatten**
With this the context view service allows placing the menu as a fixed element under the anchor element. We adopt it in the context menu service. Unfortunately, being a descendant of a series of actions and also being a series of actions comes with some interesting css quirks.

Functionally, I love this change. Implementation-wise I'd like a review since it has some clear risk and you might have ideas on how to handle parts of it better.

cc @roblourens I haven't had a chance now to look into it but I had to revert the change with the dropdowns and now it is included in this PR because the settings editor was broken. Do you mind testing the branch and taking a look? Since this is the custom select boxes, it won't repro on macOS AFAIK.

This PR fixes #90729 #93415

### CODE REVIEW

File: src/vs/base/browser/ui/menu/menu.css [hunk 1]
```diff
@@ -88,6 +88,9 @@
 	padding: 0.5em 0 0 0;
 	margin-bottom: 0.5em;
 	width: 100%;
+	height: 0px !important;
+	margin-left: .8em !important;
+	margin-right: .8em !important;
```
Comment (joaomoreno) [inline src/vs/base/browser/ui/menu/menu.css:93]: Why the `!important`?

**Review by joaomoreno — COMMENTED** [review]
This risk here also propagates to the monaco editor and it's myriad of embeddable ways. Have you tried this in the standalone editor?

**Comment by bpasero** [comment]
I glanced at the changes, but have not full understanding of the context view component to make a statement. But I think the direction makes sense to me (have the context view rooted closer to the element that triggered it). If it fixes issues and we have proper test plan item for it and Joao is fine with it, 🚢 it.

**Comment by joaomoreno** [comment]
Also, I know we're pretty confident that we don't break stuff because we have TypeScript. But often people reach out onto to the DOM and do stuff based on it. And messing around with the DOM (adding a child to a container) will break things. The tabs widget is a good example.

**Comment by sbatten** [comment]
I am scoping this down more so that less contexts are affected by the css changes. But note that this change does not always place the context view as a child of the anchor. If the menu is positioned by mouse, we will still use the top-level element. I will try to make this even more specific.

**Comment by roblourens** [comment]
It's broken in the settings editor by a couple things - we set top/left on the contextview element but it is relative to the ancestor `.monaco-list-rows` element because that has `position: relative`. Also it is a list inside a list, and the context view list is inheriting some styles from the settings list, that is fixable.

**Comment by joaomoreno** [comment]
> @joaomoreno see recent change where the caller of the contextmenuservice must opt into using fixed position and rendering as a child. this makes this change a no-op for everyone except dropdowns and select boxes meaning no effect on monaco editor.

👌

**Comment by sbatten** [comment]
Fixed the positioning of the select box in the settings editor. Turns out fixed => absolute in a translate3d context, but we can calculate the offset of the views original position to counteract that. The styling is still off because lists aren't specific enough to only style their own list items. @joaomoreno  Shadow DOM might be the only reasonably safe way forward.

**Pull request merged**